### PR TITLE
[FSTORE-1476] Deduplicate connection

### PR DESCRIPTION
**Pull request by aversey**
JIRA Issue: FSTORE-1476

Priority for Review: -

Related PRs: -

**How Has This Been Tested?**

- [x] Unit Tests
- [ ] Integration Tests
- [ ] Manual Tests on VM

**Checklist For The Assigned Reviewer:**

```
- [ ] Checked if merge conflicts with master exist
- [ ] Checked if stylechecks for Java and Python pass
- [ ] Checked if all docstrings were added and/or updated appropriately
- [ ] Ran spellcheck on docstring
- [ ] Checked if guides & concepts need to be updated
- [ ] Checked if naming conventions for parameters and variables were followed
- [ ] Checked if private methods are properly declared and used
- [ ] Checked if hard-to-understand areas of code are commented
- [ ] Checked if tests are effective
- [ ] Built and deployed changes on dev VM and tested manually
- [x] (Checked if all type annotations were added and/or updated appropriately)
```

### CODE REVIEW

File: python/hopsworks_common/connection.py [hunk 1]
```diff
@@ -359,8 +432,13 @@ def close(self) -> None:
             conn.close()
             ```
         """
+        from hsfs import engine
+
         OpenSearchClientSingleton().close()
         client.stop()
+        engine.stop()
+        self._feature_store_api = None
+        self._model_api = None
```
Comment (vatj) [inline python/hopsworks_common/connection.py:441]: model_serving_api, model_registry_api?

File: python/hopsworks_common/connection.py [hunk 2]
```diff
@@ -319,6 +377,21 @@ def connect(self) -> None:
                     hostname_verification=self._hostname_verification,
                 )
 
+            from hsfs import engine
+            from hsfs.core import feature_store_api
+
+            # init engine
```
Comment (vatj) [inline python/hopsworks_common/connection.py:383]: Why is this here and not when getting the feature store?

**Review by vatj — CHANGES_REQUESTED** [review]
Move the engine init to the get_feature_store. We don't need it until then.

Check the model serving stuff

File: python/hopsworks_common/connection.py (hunk 1 above)
Comment (aversey) [reply python/hopsworks_common/connection.py:441]: It was this way in hsml before the merge: https://github.com/logicalclocks/machine-learning-api/blob/main/python/hsml/connection.py#L200

Also, I would prefer to remove `_model_api = None` instead of adding them.

**Review by vatj — APPROVED** [review]
